webpack-build/src/index.ts

```diff
@@ -1,60 +1,117 @@
-import { AutoMapper, Mapper, MappingProfileBase } from '@nartc/automapper';
 import 'reflect-metadata';
-import { Address } from './address.model';
-import { AddressVm } from './address.vm';
-import { Profile } from './profile.model';
-import { ProfileVm } from './profile.vm';
-import { User } from './user.model';
-import { UserVm } from './user.vm';
+import { AutoMapper, Mapper, MappingProfileBase } from '@nartc/automapper';
+
+import Address from './address.model';
+import AddressVm from './address.vm';
+import Contact from './contact.model';
+import ContactVm from './contact.vm';
+import Phone from './phone.model';
+import PhoneVm from './phone.vm';
+import User from './user.model';
+import UserVm from './user.vm';
+import Gender from './gender';
+
+
+function joinNonEmpty(separator: string, ...args: Array<string | null | undefined>) {
+  return args
+    .filter(x => !!x)
+    .join(separator);
+}
 
 class UserProfile extends MappingProfileBase {
   constructor(mapper: AutoMapper) {
     super();
+
     mapper.createMap(User, UserVm)
-      .forMember(dest => dest.first, opts => opts.mapFrom(src => src.firstName))
-      .forMember(dest => dest.last, opts => opts.mapFrom(src => src.lastName))
-      .forMember(dest => dest.full, opts => opts.mapFrom(src => src.firstName + ' ' + src.lastName));
+      .forMember(d => d.fullName, o => o.mapFrom(s => joinNonEmpty(' ', s.firstName, s.middleName, s.lastName)));
   }
 }
 
-class ProfileProfile extends MappingProfileBase {
+class ContactProfile extends MappingProfileBase {
   constructor(mapper: AutoMapper) {
     super();
-    mapper.createMap(Profile, ProfileVm)
-      .forMember(dest => dest.bioString, opts => opts.mapFrom(src => src.bio))
-      .forMember(dest => dest.ageNumber, opts => opts.mapFrom(src => src.age));
+
+    mapper.createMap(Contact, ContactVm);
   }
 }
 
 class AddressProfile extends MappingProfileBase {
   constructor(mapper: AutoMapper) {
     super();
+
     mapper.createMap(Address, AddressVm)
-      .forMember(dest => dest.formattedStreet, opts => opts.mapFrom(src => src.street));
+      .forMember(d => d.street, o => o.mapFrom(s => joinNonEmpty(' ', s.line1, s.line2)));
+  }
+}
+
+class PhoneProfile extends MappingProfileBase {
+  constructor(mapper: AutoMapper) {
+    super();
+
+    // is it possible to add a map from class to simple type ex. mapper.createMap(Phone, string) ?
```

Comment:
It is not possible but regarding this case, here's another way to do it (maybe cleaner). In your `ContactVm`, have `phones` to be of type `string[]` instead of `PhoneVm[]` then utilize `convertUsing()` https://github.com/nartc/mapper#value-converter

Reply:
Hi, I've updated the code.
In my example, I have to pass an array of phones inside `PhoneConverter`. then I can use the  `convertUsing` method in the mapping profile.
But I see another approach without converters.
```javascript
export default class ContactMappingProfile extends MappingProfileBase {
    constructor(mapper: AutoMapper) {
        super();

        mapper.createMap(ContactModel, ContactViewModel)
            .forMember(
                d => d.phones,
                o => o.mapFrom(s => s.phones.map(p => stringHelper.joinNonEmpty(' x', p.phoneNumber, p.phoneExtension)))
            );
    }
}
```